addons/sale_pdf_quote_builder/tests/test_pdf_quote_builder.py

```diff
@@ -144,6 +145,23 @@ def test_dynamic_fields_mapping_for_product_document(self):
             )
             self.assertEqual(result, expected[form_field.name])
 
+    def test_product_document_dialog_params_access(self):
+        sale_order_user_demo = self.sale_order.copy({'user_id': self.user_demo.id})
+        dialog_param = sale_order_user_demo.with_user(
+            self.user_demo.id
+        ).get_update_included_pdf_params()
+        self.assertTrue(dialog_param['headers']['files'])
+        # Line product document should only be accessible by sales user
+        self.assertFalse(dialog_param['lines'])
```

Comment:
This could still work if there aren't any product documents related to this order. This `lines` could be an empty list. If not empty, then it's a list of dict containing the files, as above. Might be best to:
- ensure the product used in the order has a product document
- see if the `files` are accessible